app/src/main/java/fr/free/nrw/commons/upload/FileProcessor.java

```diff
@@ -81,6 +98,61 @@ GPSExtractor processFileCoordinates(SimilarImageInterface similarImageInterface)
         return imageObj;
     }
 
+    /**
+     * Redacts EXIF and XMP metadata as indicated in preferences.
+     *
+     */
+    @SuppressLint("CheckResult")
+    private void redactMetadata(Context context) {
+        Set<String> prefManageEXIFTags = defaultKvStore.getStringSet("manageExifTags");
+        boolean prefKeepXmp = defaultKvStore.getBoolean("keepXmp", true);
+
+        try {
+            if (!prefKeepXmp) {
+                String extension = MimeTypeMap.getFileExtensionFromUrl(Uri.fromFile(new File(filePath)).toString());
+                String newFilePath = context.getCacheDir().getAbsolutePath() + "/"
+                        + DEFAULT_FOLDER_NAME + "/"
+                        + UUID.randomUUID().toString() + "." + extension;
+
+                FileInputStream inStream = new FileInputStream(filePath);
+                FileOutputStream outStream = new FileOutputStream(newFilePath);
+                FileChannel inChannel = inStream.getChannel();
+                FileChannel outChannel = outStream.getChannel();
+                inChannel.transferTo(0, inChannel.size(), outChannel);
+                inStream.close();
+                outStream.close();
+                // Overwrites original file while removing XMP data.
+                // inputPath - newFilePath, the copy of FilePath
+                // outputPath - original FilePath
+                FileMetadataUtils.removeXmpAndWriteToFile(newFilePath, filePath);
+            }
+
+            Set<String> redactTags = new HashSet<>(Arrays.asList(
+                    context.getResources().getStringArray(R.array.pref_exifTag_values)));
+
+            Timber.d(redactTags.toString());
+            redactTags.removeAll(prefManageEXIFTags);
+
+            if (!redactTags.isEmpty()) {
+                //noinspection ResultOfMethodCallIgnored
+                Observable.fromIterable(redactTags)
```

Comment:
The observable isn't disposed. Could lead to memory leaks. Please dispose it.

Reply:
Could you help me, please? Should I call `dispose` upon `forEach` result? I thought that there's no need to `dispose()` if observable ends in its natural way.